### PR TITLE
fix sort_columns issue(12-dev)

### PR DESCRIPTION
**Pull request by QiangCai**
Fixed followed issues in sort_columns

1. for unsorted table, set taskExtension in CarbonRowDataWriterProcessorStepImpl, because one task will write more than one carbondata and carbonindex files.

2. modify getNoDictionaryRangeValues method of RowLevelRangeFilterResolverImpl to support all datatype

3. modify toBytes method of ByteUtil to support sorting of negative and positive number

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1517/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1520/

File: core/src/main/java/org/apache/carbondata/core/util/ByteUtil.java [hunk 1]
```diff
@@ -540,7 +543,7 @@ public static int toInt(byte[] bytes, int offset, final int length) {
    * @return
    */
   public static float toFloat(byte[] bytes, int offset) {
-    return Float.intBitsToFloat(toInt(bytes, offset, SIZEOF_INT));
+    return Float.intBitsToFloat(toInt(bytes, offset, SIZEOF_INT) );
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/util/ByteUtil.java]: remove space

File: core/src/main/java/org/apache/carbondata/core/scan/filter/resolver/RowLevelRangeFilterResolverImpl.java [hunk 2]
```diff
@@ -147,12 +148,16 @@ public void getStartKey(SegmentProperties segmentProperties, long[] startKey,
...
-        filterValuesList.add(result.getString().getBytes());
+        filterValuesList.add(DataTypeUtil
+            .getBytesBasedOnDataTypeForNoDictionaryColumn(result.getString(),
+                result.getDataType()));
       } catch (FilterIllegalMemberException e) {
         // Any invalid member while evaluation shall be ignored, system will log the
         // error only once since all rows the evaluation happens so inorder to avoid
         // too much log inforation only once the log will be printed.
         FilterUtil.logError(e, invalidRowsPresent);
+      } catch (Throwable throwable) {
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/scan/filter/resolver/RowLevelRangeFilterResolverImpl.java]: why is this added?

**Comment by jackylk** [comment]
Please describe the bug in JIRA or PR

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1522/

**Comment by jackylk** [comment]
please rebase

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1531/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1540/

File: core/src/main/java/org/apache/carbondata/core/scan/filter/resolver/RowLevelRangeFilterResolverImpl.java (hunk 2 above)
Comment (QiangCai) [inline core/src/main/java/org/apache/carbondata/core/scan/filter/resolver/RowLevelRangeFilterResolverImpl.java]: fixed

File: core/src/main/java/org/apache/carbondata/core/util/ByteUtil.java (hunk 1 above)
Comment (QiangCai) [inline core/src/main/java/org/apache/carbondata/core/util/ByteUtil.java]: fixed

**Comment by QiangCai** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1549/

**Comment by QiangCai** [comment]
@jackylk please review again

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1550/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1555/

**Comment by jackylk** [comment]
LGTM
